core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1472,7 +1474,9 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
       }
 
     def apply(n: Int): A =
-      if (n < start.length) start(n) else buffer(n - start.length).asInstanceOf[A]
+      if (n < 0 || n >= length) throw new IndexOutOfBoundsException(s"Append chunk access to $n")
```

Comment:
In any case where `length` is used more than once, it is best to cache it into a local variable, as a general rule in the `Chunk`.* methods.